python-package/lightgbm/sklearn.py

```diff
@@ -46,6 +46,14 @@
 )
 from .engine import train
 
+if TYPE_CHECKING:
+    # sklearn.utils.Tags can be imported unconditionally once
+    # lightgbm's minimum scikit-learn version is 1.6 or higher
+    try:
+        from sklearn.utils import Tags as _sklearn_Tags
+    except ImportError:
+        _sklearn_Tags = None
```

Comment:
I think this piece of code should go to `compat.py`.

Reply:
Yeah you're right.

(@adrinjalali , btw, https://github.com/microsoft/LightGBM/blob/master/python-package/lightgbm/compat.py is our equivalent here of what you referred to as `fixes.py` in https://github.com/microsoft/LightGBM/pull/6651#pullrequestreview-2324365084)

Reply:
I've put this stuff in `compat.py` in recent commits.